javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,373 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  private var head: Node[E] = null
+  private var last: Node[E] = head
+
+  private var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    if (isEmpty()) throw new NoSuchElementException()
+    val ret = head.value
+
+    val oldHead = head
+    head = oldHead.next
+
+    if (head ne null)
+      head.prev = null
+    else
+      head = last
```

Comment:
That would be the case if there were 2 elements before the removal and 1 after. But this case is handled (generically) in the `head ne null` case. In this case, there was 1 element prior to removal, so there will be 0 after, and that means both `head` and `last` must be `null`. There was 1 element because `oldHead.next eq null`, which is precisely the case when `oldHead` also happened to be the last element, which means there was only one element.

The following test case should break the current implementation:
```scala
val l = new LinkedList[Int]
l.addLast(1)
// now head eq last eq Node(1) and head.next eq null
l.removeFirst() // which means oldHead.next eq null here
// and now head = last which is still Node(1)
// so at this point you have a corrupted list with _size == 0 but head and last are not null
l.addLast(2) // this adds a *second* element, and _size == 1
expect(l.peekFirst()).toEqual(2) // this fails because the first element is still 1 from above
```